data_structures/src/chain.rs

```diff
@@ -160,9 +160,111 @@ impl From<Sha256> for Hash {
 pub type SHA256 = [u8; 32];
 
 /// Transaction data structure
-// FIXME(#99): define Transaction as defined in issue
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct Transaction {
+    pub version: u32,
+    pub inputs: Vec<Input>,
+    pub outputs: Vec<Output>,
+    pub claims: Vec<Claim>,
+}
+
+/// Input data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub enum Input {
+    ValueTransferInput(ValueTransferInput),
+    CommitInput(CommitInput),
+    RevealInput(RevealInput),
+    TallyInput(TallyInput),
+}
+/// Value transfer input transaction data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct ValueTransferInput {
+    pub transaction_id: [u8; 32],
+    pub output_index: u32,
+}
+
+/// Commit input transaction data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct CommitInput {
+    pub transaction_id: [u8; 32],
+    pub output_index: u32,
+    pub poe: [u8; 32],
+}
+
+/// Reveal input transaction data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct RevealInput {
+    pub transaction_id: [u8; 32],
+    pub output_index: u32,
+    pub reveal: [u8; 32],
+    pub nonce: u64,
+}
+
+/// Tally input transaction data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct TallyInput {
+    pub transaction_id: [u8; 32],
+    pub output_index: u32,
+}
+
+/// Value transfer output transaction data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct ValueTransferOutput {
+    pub pkh: Hash,
+    pub value: u64,
+}
+
+/// Data request output transaction data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct DataRequestOutput {
+    pub data_request: [u8; 32],
+    pub value: u64,
+    pub witnesses: u8,
+    pub backup_witnesses: u8,
+    pub commit_fee: u64,
+    pub reveal_fee: u64,
+    pub tally_fee: u64,
+    pub time_lock: u64,
+}
+
+/// Commit output transaction data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct CommitOutput {
+    pub commitment: Hash,
+    pub value: u64,
+}
+
+/// Reveal output transaction data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct RevealOutput {
+    pub reveal: [u8; 32],
+    pub pkh: Hash,
+    pub value: u64,
+}
+
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub struct ConsensusOutput {
+    pub result: [u8; 32],
+    pub pkh: Hash,
+    pub value: u64,
+}
+
+/// Output data structure
+#[derive(Copy, Debug, Eq, PartialEq, Clone, Serialize, Deserialize)]
+pub enum Output {
+    ValueTransferOutput(ValueTransferOutput),
```

Comment:
while accessing the variant of `Output`, we always write `Output::ValueXXX`, this already indicates `Output`, so how about trim_end the `Output` in the variant name? current version is a bit long.
```suggestion
    ValueTransfer(ValueTransferOutput),
```